[CDAP-18627] Split preview leveldb store into 2 stores.

wyzhang opened the pull request:
One for data+metadata and the other for pending

Why:
Splitting into separate levelDB stores to minimize any potential
slow seek or scan operations, since this store has high deletion
rate, thus producing lots of deletion markers or tombstones.
As a result, it is fairly easy to cause very slow seek or scan
(e.g. calling seekToStart or using iterator.next() to find target row)
during implementation. Splitting minimizes/reduces the slowness.
Furthermore, a followup PR to enable compaction on these levelDB
will allow us compacting away these deletion markers frequently,
thus speeding up access operations.

fernst reviewed (approved):
LGTM

chtyim commented on cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/preview/DefaultPreviewStore.java:
```diff
@@ -61,6 +61,7 @@
  */
 public class DefaultPreviewStore implements PreviewStore {
   private static final DatasetId PREVIEW_TABLE_ID = NamespaceId.SYSTEM.dataset("preview.table");
+  private static final DatasetId PREVIEW_PENDING_TABLE_ID = NamespaceId.SYSTEM.dataset("preview.pending.table");
```
The pending table is the queue?

wyzhang replied:
Yes.

chtyim commented on cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/preview/DefaultPreviewStore.java:
```diff
@@ -86,15 +86,18 @@
   private final AtomicLong counter = new AtomicLong(0L);
 
-  private final LevelDBTableCore table;
+  private final LevelDBTableCore dataAndMdTable;
```
so this one stores preview data? Maybe better call it `previewDataTable`?

wyzhang replied:
It stores both preview data and preview run status like start, end time etc, so more than just preview data.

chtyim replied:
So, maybe call it `previewTable` and `previewQueueTable`?

wyzhang replied:
Done.

wyzhang committed:
[CDAP-18627] Split preview leveldb store into 2 stores.
             One for data+metadata and the other for pending

Why:
Splitting into separate levelDB stores to minimize any potential
slow seek or scan operations, since this store has high deletion
rate, thus producing lots of deletion markers or tombstones.
As a result, it is fairly easy to cause very slow seek or scan
(e.g. calling seekToStart or using iterator.next() to find target row)
during implementation. Splitting minimizes/reduces the slowness.
Furthermore, a followup PR to enable compaction on these levelDB
will allow us compacting away these deletion markers frequently,
thus speeding up access operations.

wyzhang commented:
Let me know if there is any more comments, otherwise plan to merge by end of today
